presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -1210,7 +1212,8 @@ public void finishStatisticsCollection(ConnectorSession session, ConnectorTableH
         List<String> partitionColumnNames = partitionColumns.stream()
                 .map(Column::getName)
                 .collect(toImmutableList());
-        List<HiveColumnHandle> hiveColumnHandles = hiveColumnHandles(table, typeManager);
+        // TODO: revisit when handling write path
+        List<HiveColumnHandle> hiveColumnHandles = hiveColumnHandles(table, typeManager, TimestampType.DEFAULT_PRECISION);
```

Comment:
Unfortunately, I can't find an easy way to present a timestamp column with the default precision for writes (or stats collection), because the same code path is used for both read and write to get the table metadata.  I don't think that separating the paths just to ensure that writes don't fail at a precision != 3 is warranted.

Assuming we can live with this what would be a 'clean' failure?